plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -309,46 +316,197 @@ func (e *Elasticsearch) gatherClusterHealth(url string, acc telegraf.Accumulator
 			"unassigned_shards":     health.UnassignedShards,
 		}
 		acc.AddFields(
-			"elasticsearch_indices",
+			"elasticsearch_cluster_health_indices",
 			indexFields,
-			map[string]string{"index": name},
+			map[string]string{"name": name},
 			measurementTime,
 		)
 	}
 	return nil
 }
 
 func (e *Elasticsearch) gatherClusterStats(url string, acc telegraf.Accumulator) error {
-	clusterStats := &clusterStats{}
+	type ClusterStats struct {
+		ClusterName string      `json:"cluster_name"`
+		Status      string      `json:"status"`
+		Indices     interface{} `json:"indices"`
+		Nodes       interface{} `json:"nodes"`
+	}
+
+	clusterStats := &ClusterStats{}
 	if err := e.gatherJsonData(url, clusterStats); err != nil {
 		return err
 	}
 	now := time.Now()
-	tags := map[string]string{
-		"node_name":    clusterStats.NodeName,
+
+	clusterFields := map[string]interface{}{
 		"cluster_name": clusterStats.ClusterName,
 		"status":       clusterStats.Status,
 	}
 
+	acc.AddFields(
+		"elasticsearch_clusterstats",
+		clusterFields,
+		map[string]string{"name": ""},
+		now,
+	)
+
 	stats := map[string]interface{}{
 		"nodes":   clusterStats.Nodes,
 		"indices": clusterStats.Indices,
 	}
 
-	for p, s := range stats {
+	for name, stat := range stats {
 		f := jsonparser.JSONFlattener{}
 		// parse json, including bools and strings
-		err := f.FullFlattenJSON("", s, true, true)
+		err := f.FullFlattenJSON("", stat, true, true)
 		if err != nil {
 			return err
 		}
-		acc.AddFields("elasticsearch_clusterstats_"+p, f.Fields, tags, now)
+		acc.AddFields("elasticsearch_clusterstats_"+name, f.Fields, map[string]string{"name": ""}, now)
```

Comment:
Good question. This took me a while to figure out, and I'm not sure that my answer is correct. The old code had a structure to pull out the tags (node_name, cluster_name and status).  However, I ran into a problem trying to represent these as Tags on a Grafana dashboard, but when I stopped making them tags, then they represented just fine (as Strings using single-stats).  Its been too long now and I don't recall the specifics and the timing of this work. If you think it should be possible to represent them, as tags, using SingleStat - then I can certainly put them back as Tags and give it a try that way if you'd like.  Also note that node_name went away between 2.3.3 and 2.4.2, so that part of the tags/structure would no longer be possible to define.  Please let me know your thoughts.